src/app/utils/dataLoaderUtil.js

```diff
@@ -0,0 +1,158 @@
+              .map(k => ({ [decodeURIComponent(k)]: decodeURIComponent(location.query[k]) })),
+          );
+          const urlFilters = _pick(unescapedQuery, (value, key) => {
+            if (key.indexOf('filter') !== -1) {
+              return value;
+            }
+            return null;
+          });
+          Actions.updateSelectedFilters(destructureFilters(urlFilters, data.filters));
+        }
+      },
+      () => {
+        const {
+          sort,
+          sort_direction,
+        } = location.query;
+        Actions.updateSortBy(`${sort}_${sort_direction}`);
+      },
+    ],
+    errorMessage: 'Error attempting to make an ajax request to search',
+  },
+  holdRequest: {
+    apiRoute: (matchData, route) => route.replace(':bibId-:itemId', matchData[1]),
+    actions: [
+      data => Actions.updateBib(data.bib),
+      data => Actions.updateDeliveryLocations(data.deliveryLocations),
+      data => Actions.updateIsEddRequestable(data.isEddRequestable),
+    ],
+    errorMessage: 'Error attempting to make ajax request for hold request',
+  },
+});
+
+// const reducePathExpressions = location => (acc, instruction) => {
+//   const {
+//     pathname,
+//     search,
+//   } = location;
+//
+//   const {
+//     matchData,
+//     pathType,
+//   } = acc;
+//
+//   const newMatchData = (pathname + search).match(instruction.expression);
+//   return {
+//     matchData: matchData || newMatchData,
+//     pathType: pathType || (newMatchData ? instruction.pathType : null),
+//   };
+// };
+
+const matchingPathData = (location) => {
+  const {
+    pathname,
+    search,
+  } = location;
+
+  return pathInstructions
+    .map(instruction => ({
+      matchData: (pathname + search).match(instruction.expression),
+      pathType: instruction.pathType,
+    }))
+    .find(pair => pair.matchData)
+    || { matchData: null, pathType: null };
+};
+
+function loadDataForRoutes(location, req, routeMethods) {
+  const routes = routesGenerator(location);
+  const {
+    matchData,
+    pathType,
+  } = matchingPathData(location);
+
+  if (routes[pathType]) {
+    const {
+      apiRoute,
+      actions,
+      errorMessage,
+    } = routes[pathType];
+    const route = routePaths[pathType];
+    Actions.updateLoadingStatus(true);
+    const successCb = (response) => {
+      actions.forEach(action => action(response.data));
+      Actions.updateLoadingStatus(false);
+    };
+    const errorCb = (error) => {
+      Actions.updateLoadingStatus(false);
+      console.error(
+        errorMessage,
+        error,
+      );
+    };
+    if (req) {
+      return new Promise((resolve) => {
+        const res = {
+          json: (data) => {
+            resolve({ data });
+          },
+        };
+        routeMethods[pathType](req, res);
+      })
+        .then(successCb)
+        .catch(errorCb);
+    }
+    return ajaxCall(apiRoute(matchData, route), successCb, errorCb);
+  }
+
+  return new Promise(resolve => resolve());
```

Comment:
This seems okay for now, but maybe you want to reject and throw an error if no routes match?

Reply:
My thinking was that `dataLoaderUtil` only needs to know about routes that actually require data fetching. So if there are no instructions to fetch data for a particular route, it isn't an error, just a no-op.

Reply:
Yeah, that makes sense. It just seems odd returning an empty resolved promise. There doesn't seem to be a good alternative.